src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((acc, item) => {
+    const keyValue = item[key] as unknown as string;
```

Comment:
Casting `item[key]` directly to `string` can lead to runtime errors if `item[key]` is not a string. Consider verifying the type before casting or handling the potential type mismatch more gracefully.